src/catch2/internal/catch_run_context.cpp

```diff
@@ -319,6 +319,16 @@ namespace Catch {
         return tracker;
     }
 
+    void RunContext::trackGeneratorState( GeneratorInfo const& info ) {
+        // Avoid redundant entries, in case a generator is used within a loop.
+        if ( std::find( m_generatorInfos.cbegin(),
```

Comment:
Since `info.name` is already unique, we could use `find_if` based on the name only. Saves extra some comparisons.